### PR TITLE
feat(regularisation) : #MA-470 add feedback on absence regularisation

### PR DESCRIPTION
**Pull request by Alex-Delaveau**
## Describe your changes
Add a confirmation toaster on absence regularisation.
image

## Checklist tests
Add an absence that is regularizable, then go to the dashboard side panel, regularize it, and check if you receive a confirmation feedback.
## Issue ticket number and link
[MA-470](https://jira.support-ent.fr/browse/MA-470)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by flmartineau — APPROVED** [review]

File: presences/src/main/resources/public/ts/models/CounsellorAbsence.ts [hunk 1]
```diff
@@ -41,14 +41,16 @@ export class CounsellorAbsence {
...
+    async updateRegularisation() : Promise<boolean> {
         try {
             await http.put('/presences/absence/regularized', {
                 regularized: this.counsellor_regularisation,
                 ids: [this.id]
-            });
+            })
+            return true;
         } catch (err) {
             notify.error('presences.absences.update_regularized.error');
```
Comment (qpic-cgi) [inline presences/src/main/resources/public/ts/models/CounsellorAbsence.ts]: tu vois, tu peux peut-être gérer ton feedback ici du coup
(étant donné qu'il est déjà géré dans le catch comme ça.

File: presences/src/main/resources/public/ts/controllers/widgets/absences.ts [hunk 2]
```diff
@@ -254,10 +254,18 @@ export const absencesController = ng.controller('AbsenceController', ['$scope',
...
+            let res;
             if (absence instanceof CounsellorAbsence) {
-                await absence.updateRegularisation();
+                res = await absence.updateRegularisation();
             } else {
-                await new Events().updateRegularized([absence], absence.counsellor_regularisation, absence.student.id, window.structure.id);
+                res = await new Events().updateRegularized([absence], absence.counsellor_regularisation, absence.student.id, window.structure.id)
+            }
+            if(res){
+                toasts.confirm('presences.absences.update_regularized')
```
Comment (qpic-cgi) [inline presences/src/main/resources/public/ts/controllers/widgets/absences.ts]: Je crois que désormais on utilise notify (et plus toast).
A voir avec ta PO.

File: presences/src/main/resources/public/ts/models/Event/Event.ts [hunk 3]
```diff
@@ -317,13 +317,15 @@ export class Events extends LoadingCollection {
...
-    async updateRegularized(events, regularized, studentId, structureId): Promise<void> {
+    async updateRegularized(events, regularized, studentId, structureId): Promise<boolean> {
         if (events.length === 0) return;
         try {
             events.forEach(e => delete e.$$hashKey);
             await http.put(`/presences/events/regularized`, {events: events, regularized: regularized, student_id: studentId, structure_id: structureId});
+            return true
         } catch (err) {
             throw err;
+            return false;
```
Comment (qpic-cgi) [inline presences/src/main/resources/public/ts/models/Event/Event.ts:331]: pareil que les coms plus haut

**Review by qpic-cgi — CHANGES_REQUESTED** [review]

File: presences/src/main/resources/public/ts/controllers/widgets/absences.ts (hunk 2 above)
Comment (Alex-Delaveau) [reply presences/src/main/resources/public/ts/controllers/widgets/absences.ts]: J'ai vu avec Celia toast est mieux et c'est celui que ode veulent garder sur le long terme apparement.

File: presences/src/main/resources/public/ts/controllers/widgets/absences.ts [hunk 4]
```diff
@@ -254,11 +254,16 @@ export const absencesController = ng.controller('AbsenceController', ['$scope',
          * @param absence absence an absence, can be an absence event.
          */
         vm.updateRegularisationAbsenceEvent = async (absence: CounsellorAbsence | IEvent): Promise<void> => {
+            let res;
```
Comment (qpic-cgi) [inline presences/src/main/resources/public/ts/controllers/widgets/absences.ts]: type

**Review by qpic-cgi — CHANGES_REQUESTED** [review]

**Review by qpic-cgi — APPROVED** [review]

**Pull request merged**